python/lsst/meas/base/baseMeasurement.py

```diff
@@ -280,8 +280,13 @@ def initializePlugins(self, **kwds):
         # Initialize the plugins to run on the undeblended image
         for executionOrder, name, config, PluginClass in sorted(self.config.undeblended.apply()):
             undeblendedName = self.config.undeblendedPrefix + name
-            self.undeblendedPlugins[name] = PluginClass(config, undeblendedName, metadata=self.algMetadata,
-                                                        **kwds)
+            if hasattr(PluginClass, "hasLogName") and PluginClass.hasLogName:
```

Comment:
Looking at this now, I don't like the duplication of the complex arguments below and I don't like the code this was copied from above. I think I would much prefer it to be something like:

```python
logName = None
if hasattr(PluginClass, "hasLogName") and PluginClass.hasLogName:
    logName = self.log.getChild(name).name
self.undeblendedPlugins[name] = PluginClass(config, undeblendedName, metadata=self.algMetadata,
                                            logName=logName, **kwds)
```
Then it is clear that the `if`-statement is solely modifying the `logName` parameter and it doesn't force you to look at the other parameters to ensure they haven't been changed in the two cases. Please fix the code above in the same way.

Reply:
Actually, I think this could be even cleaner:
```python
if getattr(PluginClass, "hasLogName", False):
    logName = self.log.getChild(name).name
self.undeblendedPlugins[name] = PluginClass(config, undeblendedName, metadata=self.algMetadata,
                                            logName=logName, **kwds)
```
Maybe I've missed something obvious though since I'm a little tired at the moment.

Reply:
I think this logic is completely taken care in `wrappers.py` and we might be able to always pass in a `logName` unconditionally. Let me try that out tomorrow.

Reply:
Hey Tim, this won't work because not all plugins have `logName` as one of their arguments. So there will have to be two different calls, depending on whether the `__init__` of the `PluginClass` can take `logName` as a keyword argument or not. We can certainly simplify the logic though. Or a more painful exercise is to make all the plugins in our codebase to accept it as an optional argument.